src/KeyringController.test.ts

```diff
@@ -834,6 +867,19 @@ describe('KeyringController', () => {
       // removing the account since it still has an account left
       expect(keyringController.keyrings).toHaveLength(2);
     });
+
+    it('throws an error if the keyring for the given account does not support account removal', async () => {
+      const mockAccount = '0x5aC6d462f054690A373Fabf8cc28E161003aEB19';
+      stubKeyringClassWithAccount(BaseKeyringMock, mockAccount);
+      const keyringController = await initializeKeyringController({
+        password: PASSWORD,
+      });
```

Comment:
I don't know if this is a pre-existing pattern in these tests, but there seems to be a disconnect in the described behavior, the data that the test is using, and the place where that data is defined. This test seems to be relying on the fact that BaseKeyringMock is the default keyring for all tests. Defaults are generally useful for data that doesn't matter, but in this case the keyring _does_ matter, as the test will only pass when using a keyring that doesn't support account removal.

Maybe just to be absolutely certain that the test does the right thing, it would be useful to pass BaseKeyringMock as the keyring used?

``` suggestion
      stubKeyringClassWithAccount(BaseKeyringMock, mockAccount);
      const keyringController = await initializeKeyringController({
        password: PASSWORD,
        constructorOptions: {
          keyringBuilders: [keyringBuilderFactory(BaseKeyringMock)],
        }
      });
```

That way it's easier to see what the controller is using and why the error below is produced.

Again, though, if this would require refactoring these tests to do, don't worry about it — just something that might improve the readability of these tests in the future.

Reply:
Good point, the test relies on `BaseKeyringMock` as the default "additional" keyring during tests (in the `initializeKeyringController` function), but this might be a hidden behavior that makes the test case less readable.

I applied your suggestion in 988974e